OpenRA.Mods.Common/Widgets/Logic/MainMenuLogic.cs

```diff
@@ -274,19 +276,48 @@ void LoadAndDisplayNews(WebServices webServices, Widget newsBG)
 				{
 					if (!fetchedNews)
 					{
-						// Send the mod and engine version to support version-filtered news (update prompts)
-						var newsURL = "{0}?version={1}&mod={2}&modversion={3}".F(
-							webServices.GameNews,
-							Uri.EscapeUriString(Game.EngineVersion),
-							Uri.EscapeUriString(Game.ModData.Manifest.Id),
-							Uri.EscapeUriString(Game.ModData.Manifest.Metadata.Version));
-
-						// Parameter string is blank if the player has opted out
-						newsURL += SystemInfoPromptLogic.CreateParameterString();
-
-						new Download(newsURL, cacheFile, e => { },
-							e => NewsDownloadComplete(e, cacheFile, currentNews,
-								() => OpenNewsPanel(newsButton)));
+						Task.Run(async () =>
+						{
+							try
+							{
+								var client = HttpClientFactory.Create();
+
+								// Send the mod and engine version to support version-filtered news (update prompts)
+								var url = new QueryBuilder(webServices.GameNews)
+								{
+									{ "version", Game.EngineVersion },
+									{ "mod", Game.ModData.Manifest.Id },
+									{ "modversion", Game.ModData.Manifest.Metadata.Version }
+								}.ToString();
+
+								// Parameter string is blank if the player has opted out
+								url += SystemInfoPromptLogic.CreateParameterString();
+
+								var response = await client.GetStringAsync(url);
```

Comment:
`response` is unused.
You need to add `File.WriteAllText(cacheFile, response);` or similar (you could do it like the package download code if you like).